### ISSUE TITLE
Store geojson locally

### ISSUE THREAD

**Issue by nathanhollows** [nathanhollows/NuggetNav#7]
To ensure the app is usable straight away without a wifi connection the location data should be stored locally and then cached in future. The location data is from [nuggetwatch.co.nz/locations](nuggetwatch.co.nz/locations)

Here's how it should work:

- App is shipped with `locations` or 'locations.json`
- On launch, if wifi is available then replace `locations` with nuggetwatch.co.nz/locations
- All components should check the same location for the json file, preferably the cache. If nothing is found then use the shipped copy

**Comment by nathanhollows** [comment]
These lines in `MapActivity` set fetch the locations. MapBox automatically fetches it using the URI, but it would also accept a string instead. The `okhttp` library might be a simple way to fetch and cache this.

```java
// Add a new source from the GeoJSON data and set the 'cluster' option to true.
try {
    source = new GeoJsonSource("locations",
            new URI("https://nuggetwatch.co.nz/locations"), // this line here
            new GeoJsonOptions()
                    .withCluster(true)
                    .withClusterMaxZoom(14)
                    .withClusterRadius(50)
    );
    loadedMapStyle.addSource(source);
} catch (URISyntaxException uriSyntaxException) {
}
```

[MapBox docs for GeoJsonSource](https://docs.mapbox.com/android/api/map-sdk/5.1.2/com/mapbox/mapboxsdk/style/sources/GeoJsonSource.html)

**Comment by nathanhollows** [comment]
ee9f0cd stores the locations file in a cache for 24 hours. I'm not sure if it holds it indefinitely if the phone isn't internet connected. In case the connection fails it should still pull from a local copy if there is nothing else available

**Comment by nathanhollows** [comment]
There's a race condition when loading the locations json file and building the map. The app occasionally loads the file first and tries to add it to the map before it is fully initialised. It can cause a crash on start up

**Comment by nathanhollows** [comment]
Solved with 724b01311b52dcda68084108586723d72639eaee
